language/functional-tests/src/compiler.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use bytecode_verifier::verifier::{
+    verify_module_dependencies, verify_script_dependencies, VerifiedModule, VerifiedScript,
+};
+use libra_types::{account_address::AccountAddress, vm_error::VMStatus};
+use vm::file_format::{CompiledModule, CompiledScript};
+
+pub trait Compiler {
+    type ParsedScript;
+    type ParsedModule;
+
+    /// Compile a transaction script or module.
+    fn compile<Logger: FnMut(String) -> ()>(
+        &mut self,
+        log: Logger,
+        address: AccountAddress,
+        input: &str,
+    ) -> Result<ScriptOrModule>;
+
+    /// Verifies a script and its dependencies.
+    fn verify_script(
```

Comment:
Add comment "// TODO: Verification should not be part of the compilation pipeline. Will remove in the future." or similar.

Reply:
I might be able to remove it actually, let me take another look